sphinx_gallery/directives.py

```diff
@@ -58,6 +58,14 @@ def run(self):
             heading_level = self.options.get('heading-level', '^')
             lines.append(heading_level * len(heading))
 
+        def should_show(obj):
+            # TODO: get the path in a cleaner and more robust way
+            path = os.path.abspath(os.path.join("source", backreferences_dir, '{}.examples'.format(obj)))
```

Comment:
We populate `sphinx_gallery_conf` with extra values:

https://github.com/sphinx-gallery/sphinx-gallery/blob/master/sphinx_gallery/gen_gallery.py#L161

One of these is `src_dir`, so you should be able to do something like:
```suggestion
            src_dir = config.sphinx_gallery_conf['src_dir']
            path = os.path.abspath(os.path.join(src_dir, backreferences_dir, '{}.examples'.format(obj)))
```